src/chip.cs

```diff
@@ -1132,54 +1548,148 @@ interface ChipOnOff
 	[DisableDefaultCtor]
 	interface ChipOperationalCredentials
 	{
-
+		[Mac (12,1), Watch (8,3), TV (15,2), iOS (15,2), MacCatalyst (15,2)]
 		[Export ("initWithDevice:endpoint:queue:")]
 		[DesignatedInitializer]
+		NativeHandle Constructor (ChipDevice device, ushort endpoint, DispatchQueue queue);
+
+#if !NET
+		[Deprecated (PlatformName.iOS, 15, 2)]
+		[Deprecated (PlatformName.TvOS, 15, 2)]
+		[Deprecated (PlatformName.WatchOS, 8, 3)]
+		[Deprecated (PlatformName.MacOSX, 12, 1)]
+		[Deprecated (PlatformName.MacCatalyst, 15, 2)]
+		[Wrap ("this (device, (ushort) endpoint, queue)")]
 		NativeHandle Constructor (ChipDevice device, byte endpoint, DispatchQueue queue);
+#endif
 
+		[Obsolete ("This method is removed.")]
+		[Wrap ("throw new InvalidOperationException (Constants.RemovedFromChip)", IsVirtual = true)]
 		[NoMac] // fails on macOS 12 beta 6
 		[Async (ResultTypeName = "ChipReadAttributeResult")]
-		[Export ("addOpCert:iCACertificate:iPKValue:caseAdminNode:adminVendorId:responseHandler:")]
 		void AddOpCert (NSData noc, NSData iCACertificate, NSData iPKValue, ulong caseAdminNode, ushort adminVendorId, ChipResponseHandler responseHandler);
 
+		// Parameter names are left to match header files. Without documentation, we cannot know what the parameters signify for certain.
+		[Mac (12,1), Watch (8,3), TV (15,2), iOS (15,2), MacCatalyst (15,2)]
+		[Async (ResultTypeName = "ChipReadAttributeResult")]
+		[Export ("addNOC:iPKValue:caseAdminNode:adminVendorId:responseHandler:")]
+		void AddNOC (NSData nOCArray, NSData iPKValue, ulong caseAdminNode, ushort adminVendorId, ChipResponseHandler responseHandler);
```

Comment:
```suggestion
		void AddNoc (NSData nOCArray, NSData iPKValue, ulong caseAdminNode, ushort adminVendorId, ChipResponseHandler responseHandler);
```

Reply:
should I also switch nOCArray to nocArray then?